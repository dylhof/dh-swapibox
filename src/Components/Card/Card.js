import React from 'react';

const Card = (props) => {
  let card;
  
  switch (props.category) {
    case 'person':
      card =
        <div>
          <h3 >Name: {props.name} <i class="fas fa-star"></i></h3>
          <p>Homeworld: {props.homeworld}</p>
          <p>Species: {props.species}</p>
          <p>Homeworld Population: {props.population}</p>
          
        </div>
      break;
    case 'planet': 
      card = 
        <div>
          <h3>Name: {props.name} <i class="fas fa-star"></i></h3>
          <p>Terrain: {props.terrain}</p>
          <p>Climate: {props.climate}</p>
          <p>Residents:</p>
          <ul> 
            {props.residents.map(resident => {
              return <li key={resident}>{resident}</li>
            })}
          </ul>
        </div>
      break;
    case 'vehicle':
      card = 
        <div>
          <h3>Name: {props.name} <i class="fas fa-star"></i></h3>
          <p>Model: {props.model}</p>
          <p>Class: {props.class}</p>
          <p>Passengers: {props.passengers}</p>
        </div>
      break;
  }

  return (
    <div className='card'>{card}</div>
  )
}

export default Card;