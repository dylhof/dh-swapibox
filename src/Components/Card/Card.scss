.card {
  background-color: white;
  color: black;
  height: 200px;
  width: 200px;
}