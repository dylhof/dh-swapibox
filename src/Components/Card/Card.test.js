import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';

describe('Card', () => {
  let wrapper;
  let cardDataMock
  beforeEach(() => {
    wrapper = shallow(<Card card={cardDataMock}/>)
  })
  it('should match the snapshot if category is person', () => {
    //setup
    cardDataMock = {name: 'luke', homeworld: 'tatoine', species: 'human', population: 20, category: 'person', id: 1}
    //execution && expectation
    expect(wrapper).toMatchSnapshot();
  })

  it('should match the snapshot if category is planet', () => {
    //setup
    cardDataMock = {name: 'tatoine', terrain: 'desert', climate: 'hot', residents: ['luke', 'Beru'], category: 'planet', id: 1}
    //execution && expectation
    expect(wrapper).toMatchSnapshot();
  })

  it('should match the snapshot if category is vehicle', () => {
    //setup
    cardDataMock = {name: 'At-At', model: '5', class: 'red', passengers: 30, category: 'vehicle', id: 1}
    //execution && expcetation 
    expect(wrapper).toMatchSnapshot();
  })
})