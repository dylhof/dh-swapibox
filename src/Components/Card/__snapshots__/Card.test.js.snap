// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot if category is person 1`] = `
<div
  className="card"
/>
`;

exports[`Card should match the snapshot if category is planet 1`] = `
<div
  className="card"
/>
`;

exports[`Card should match the snapshot if category is vehicle 1`] = `
<div
  className="card"
/>
`;
